port_ocean/core/handlers/entities_state_applier/port/applier.py

```diff
@@ -33,35 +30,40 @@ class HttpEntitiesStateApplier(BaseEntitiesStateApplier):
     """
 
     async def _validate_delete_dependent_entities(self, entities: list[Entity]) -> None:
-        logger.info("Validated deleted entities")
-        if not event.port_app_config.delete_dependent_entities:
-            dependent_entities = await asyncio.gather(
-                *(
-                    self.context.port_client.search_dependent_entities(entity)
-                    for entity in entities
-                )
-            )
-            new_dependent_entities = get_unique(
-                [
-                    entity
-                    for entity in chain.from_iterable(dependent_entities)
-                    if not any(is_same_entity(item, entity) for item in entities)
-                ]
+        if event.port_app_config.delete_dependent_entities:
+            return
+
+        logger.info("Validating no dependent entities blocks the operation")
+        dependent_entities = await asyncio.gather(
+            *(
+                self.context.port_client.search_dependent_entities(entity)
+                for entity in entities
             )
```

Comment:
This is memory and context heavy on the running program, as you can spawn thousands of search at ones. And also on the API of port. Consider using asyncio queue or semaphore